### DISCUSSION TITLE
Rare behaviour of better tooltip feature in combination with macro used through mkdocs-macros plugin

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#6462] [Q&A]
I encountered an interesting behaviour with the better tooltip feature of Material for MkDocs.

I use the macros plugin for MkDocs to insert macros I made. One such macro is named `game.info` and inserts an inline admonition box with a table containing info...
The content is also send through a `markdown.Markdown(...)` instance to parse markdown and apply changes from extensions.

One extension used is abbr to add abbreviations. What is now interesting is the fact that the abbreviation is applied to the matching text - which in this case is a version, which should display its release date - but no title is displayed anymore on hover. Not with the better tooltip, nor with the default title from the browser.

Is there any possible fix here to get better tooltip to work here, or at the very least have the default title displayed again?

For some extra context, the markdown instance I mentioned is done in a separate macro, so that I can reuse it where necessary.
Here is the simple code:
```py
    @env.macro
    def markdownify(text: str):
        md = markdown.Markdown(
            extensions=env.conf['markdown_extensions'],
            extension_configs=env.conf['mdx_configs'] or {}
        )
        # markdown adds <p> tags, which we don't need here.
        return md.convert(text).replace('<p>', '').replace('</p>', '')
```
I assume I may need to somehow apply mkdocs-material to the Markdown instance to have the feature applied properly?

**Comment by Andre601** [comment]
Another weird behaviour:
The better tooltip shows up at the wrong location. Shown here where I hovered over the text with the abbreviation (Highlighted in yellow) and the hover appears further to the right than it should (Highlighted in light blue)
![grafik](https://github.com/squidfunk/mkdocs-material/assets/11576465/60ffdf0b-d3b3-4ee1-9d65-3d9c9491bf73)

Could this be because of the line break the text has?
Also, to clarify is this not the text I mentioned in my first post. This one is unrelated.

**Comment by Andre601** [comment]
Looking a bit more into this, the tooltips seem to all exist, but are at completely wrong locations.

I would really love for some input on what could be the issue here and if there is a possible fix.

**Comment by Andre601** [comment]
I want to mention that I still do have this problem, and I cannot quite get why it happens the way it does here... So any ideas on what the cause might be and possible fixes - if any - would be greatly apreciated.

**Reply by kamilkrzyskow** [reply]
A similar bug is being investigated in https://github.com/squidfunk/mkdocs-material/issues/6488 
would be great if you could provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) that doesn't rely on the macros plugin as the better tooltips are generated in JavaScript after the page was rendered and their position is also calculated on the fly.
Perhaps your page is zoomed in or zoomed out and hence the position is incorrect or calculating the x/y position of the inline text is rather complicated.

**Reply by Andre601** [reply]
I believe the issue is caused with the macros plugin itself somehow...
Or perhaps it may be an issue related to using [inline admonition blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#inline-blocks) here.
My final guess is the fact I'm adding HTML and not markdown through the macros, and that this messes things up.

While I doubt a minimal reproduction may replicate the issue will I still try it...

**Reply by Andre601** [reply]
Okay, it seems the tooltips calculation doesn't aknowledge the inline admonition block at the end, causing the text with the abbreviation to split. Will send my reproduction to the issue with the additional info.
